src/main/java/io/confluent/idesidecar/restapi/auth/RefreshCCloudTokensBean.java

```diff
@@ -21,8 +21,12 @@ public class RefreshCCloudTokensBean {
   private static final Integer MAX_TOKEN_REFRESH_ATTEMPTS = ConfigProvider.getConfig()
       .getValue("ide-sidecar.connections.ccloud.refresh-token.max-refresh-attempts", Integer.class);
 
+  final ConnectionStateManager connectionStateManager;
+
   @Inject
-  ConnectionStateManager connectionStateManager;
+  RefreshCCloudTokensBean(ConnectionStateManager connectionStateManager) {
+    this.connectionStateManager = connectionStateManager;
+  }
```

Comment:
This change is needed so that we can pass a manually-created mock of the `ConnectionStateManager` when testing this bean, which seems to fix some flaky tests (see https://github.com/confluentinc/ide-sidecar/pull/199#issuecomment-2511388377).